both/api/sources/server/stream-chain/stream-types/transform-data/index.js

```diff
@@ -0,0 +1,98 @@
+import { check } from 'meteor/check';
+import { SimpleSchema } from 'meteor/aldeed:simple-schema';
+import { _ } from 'lodash';
+import entries from '/both/lib/entries';
+import getContext from './get-context';
+import vm from 'vm';
+
+const { Transform } = Npm.require('zstreams');
+
+function compileMapping(fieldName, javascript, context) {
+  const code = `(d) => (${javascript})`;
+
+  return vm.runInContext(code, context);
```

Comment:
Hmm, this gives us a reference to the compiled function, but can't the function access its prototype at runtime, allowing it to change stuff outside the VM? We can test this later. If this should be the case, we can still switch the pattern to `const script = new vm.Script(…); script.runInContext(context)` and return the values by changing the context, JSONifying it and then de-JSONifying it again.